worker-core/src/main/java/com/hpe/caf/worker/core/WorkerTaskImpl.java

```diff
@@ -143,17 +143,44 @@ private TaskMessage createResponseMessage(final boolean includeTaskContext, fina
 
         final String responseMessageType = response.getMessageType();
 
+        //  Check if a tracking change is required.
+        final TrackingInfo trackingInfo;
+        if (null != response.getTrackTo()) {
```

Comment:
I believe this code can be simplified by checking for empty string first rather than `null`.